Update BusquedaLocal.java

jabm0010 opened the pull request:
Lo que he hecho hoy, he comenzado a plantear un poco el algoritmo de búsqueda local aunque todavía me queda terminarlo, te lo paso por si se te ocurre a ti como seguirlo

jabm0010 committed:
Update BusquedaLocal.java

jabm0010 committed:
Cambios en BusquedaLocal

He cambiado lo que puse en el anterior commit, he tratado de explicar todo lo que he hecho en los comentarios del código, a priori creo que lo que es la base del algoritmo debería de funcionar, me falta aplicar la factorización para la solución

jabm0010 committed:
Update BusquedaLocal.java

jabm0010 committed:
Update main.java

jabm0010 reviewed (commented):
Añadida factorización
Lo he probado y tengo problemas de out of range

PierreSimT commented:
Donde te da el out_of_range ?

PierreSimT commented:
La busqueda local ya funciona, ahora subo como tengo el resultado final, aunque con 10000 iteraciones tarda bastante hahah
Ya con 100 iteraciones dura unos 40 - 50 segundos hasta que termine 

EDIT: Ya lo he subido

jabm0010 committed:
Update BusquedaLocal.java

Terminada mi versión de busqueda local. Es bastante lenta pero parece funcionar

jabm0010 committed:
Mejora optimización, greedy y menu

Cambios que he hecho:
-Clase restricciones, para evitar tener que hacer la lectura del archivo en cada ciclo he creado una clase restricción donde se guardan los datos en una List<List<Integer>> , mejorando el tiempo de ejecución del algoritmo
-Búsqueda local, he hecho los ajustes necesarios, la lógica sigue siendo la misma pero ahora en vez de leer desde el fichero para las restricciones coge los datos directamente
-Greedy, algoritmo greedy aleatorio, como lo piden como un algoritmo aparte en la práctica lo he montado ya para tenerlo en un .java diferente
-En el main he añadido un menu para hacer más accesible el uso cuando queramos probar varios algoritmos

jabm0010 committed:
Add files via upload

jabm0010 committed:
Add files via upload

búsquedaTabú y un main para probarla.
He subido tambíen un documento de texto explicando un poco lo que he hecho en la busquedaTabu.

Hay que arreglar en la búsqueda tabú errores de out of bound

jabm0010 committed:
Terminada Busqueda Tabu

Algoritmo de búsqueda tabú, parece funcionar de forma correcta. Comprobado además que funciona en caso de estancamiento de 2000 iteraciones.

jabm0010 committed:
Algoritmo Grasp

He hecho el proceso de creación de la tabla, basicamente a partir de los valores de coste le asigno un valor posición correspondiente a cada combinación transmisor-frecuencia que hemos obtenido del greedy. A partir de ahí, saco un valor probabilístico, y finalmente le asocio a todos un intervalo entre 0-1 en función de esa probabilidad. Con un aleatorio se elige un intervalo y los valores de transmisor y frecuencia que estaban asociados a él son los que me quedo.

jabm0010 committed:
Update Grasp.java

jabm0010 committed:
Update Grasp.java

Funciona la creación de tabla y elección de transmisor-frecuencia

jabm0010 committed:
Añadida opción para cambiar semilla
